Update all the things (except EF Core dependencies)

dougbu opened the pull request:
- see #9923 for related discussion
- pin aspnet/EntityFrameworkCore dependencies
- chain aspnet/Extensions dependencies through aspnet/AspNetCore-Tooling instead
- run `darc` to grab the latest

nits:
- copy a few `ReferenceResolution.md changes from aspnet/Extensions version

dougbu committed:
Update all the things (except EF Core dependencies)
- see #9923 for related discussion
- pin aspnet/EntityFrameworkCore dependencies
- chain aspnet/Extensions dependencies through aspnet/AspNetCore-Tooling instead
- run `darc` to grab the latest

nits:
- copy a few `ReferenceResolution.md changes from aspnet/Extensions version

natemcmaster commented:
FYI @Eilon @ajcvickers this freezes our dependency on EFCore indefinitely.

@dougbu if you haven’t already, can you open an issue to make sure we revisit this before Preview 6 is complete? Pausing is ok as a temporary measure, but if EFCore hasn’t stabilized in a few weeks, we will need to raise the urgency on getting fixes.

BrennanConroy commented on docs/ReferenceResolution.md:
```diff
@@ -81,3 +96,13 @@ If Microsoft.AspNetCore.Banana in 2.1 had a reference to `Microsoft.AspNetCore.O
+2. Run `darc update-dependencies --channel '.NET Core 3 Dev'`
+   * Use `'.NET Core 3 Release'` in a `release/3.0-*` branch
+3. `git diff` to confirm the tool's changes
+4. Proceed with usual commit and PR&hellip;
```
```suggestion
4. Proceed with usual commit and PR
```

dougbu commented:
> can you open an issue to make sure we revisit this before Preview 6 is complete?

Filed aspnet/AspNetCore#10050

> Suggested change&hellip;

I like horizontal ellipses (&hellip;) but 🆗 😸 I'll combine that with taking care of System.Text.Encodings.Web now being in the runtime packs for Microsoft.NetCore.App.

dougbu committed:
Address PR comments and fix Shared FX tests
- System.Text.Encodings.Web now included in runtime.*.Microsoft.NETCORE.App packages

dougbu committed:
Remove remaining System.Text.Encodings.Web references
- left the mentions in Version.Details.xml and Versions.props because they don't hurt anything
